common/src/ui/DrawShapeToolController2D.h

```diff
@@ -44,6 +44,9 @@ class DrawShapeToolController2D : public ToolController
   const Tool& tool() const override;
 
   std::unique_ptr<GestureTracker> acceptMouseDrag(const InputState& inputState) override;
+  
+  // Selection
+  std::unique_ptr<GestureTracker> handleBoxSelection(const InputState& inputState);
```

Comment:
`DrawShapeToolController` is the wrong place for this. This would require refactoring `SelectionTool` to split out separate 2D and 3D controllers, then extending the 2D controller to handle the box selection.